Abort on failed system call

andyw8 opened the pull request:
### WHY are these changes introduced?

As seen in #1026, if running a Rails generator command fails, an error is outputted but the script still continues. This can lead to the command appearing to succeed even though a broken/incomplete app was generated.

I've also opened https://github.com/Shopify/cli-kit/pull/123 to fix some some incorrect documentation.

### WHAT is this pull request doing?

Check the process status of system calls and abort if it fails.

### Update checklist
<!--
  Ideally, CHANGELOG entries should be in the format
  `* [#PR](PR URL): Message`. You should consider adding your PR
  and then making the CHANGELOG update once you know the PR number.
-->
- [x] I've added a CHANGELOG entry for this PR (if the change is public-facing)

andyw8 commented on lib/shopify-cli/context.rb:
```diff
@@ -336,7 +336,11 @@ def uname
+      unless process_status.success?
+        abort("System call failed: #{args.join(' ')}")
+      end
+      true
```
Another approach here would have been to return `process_status.success` and have the caller responsible for checking the result, but I feel this way is safer.

paulomarg replied:
I agree, if we can rescue the `abort` call it's safer to always fail unless explicitly handling errors.

andyw8 commented on test/project_types/rails/commands/create_test.rb:
```diff
@@ -45,7 +45,7 @@ def test_can_create_new_app
         expect_command(%W(#{gem_path}/bin/bundle install),
                        chdir: File.join(@context.root, 'test-app'))
-        expect_command(%W(#{gem_path}/bin/spring stop),
+        expect_command(["spring stop || true"],
```
(I don't fully understand why the previous test included the `gem_path`).

mkevinosullivan replied:
If I remember correctly, it has to do with Windows.  GEM_HOME is usually set in the environment for mac (especially our environments) but on Windows, one has to figure it out (by retrieving output from `gem environment home`).

Hence `Gem.binary_path_for()` (used by `syscall()` in `rails/commands/create.rb`) will default to just the binary name on mac but on Windows will be a fullpath name, picking up GEM_HOME as derived from `gem environment home`.

andyw8 commented:
I changed the `system` method to return a boolean, but I've now changed it back to returning the `Process::Status` because I noticed some existing code (e.g. in themekit.rb) is checking for `.success?`.
